Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,578 @@
+
+The library supports routing for common OData endpoints like entity sets, singletons, actions, functions, nested properties, etc. You can specify an endpoint for a specific controller action explicitly using ASP.NET Core `[Route]` attributes. You can also define your own routing conventions to extend the built-in conventions or define custom routing logic.
+
+Visit the [routing documentation](/odata/webapi-8/fundamentals/routing-overview) to learn more about routing.
+
+## Query handling
+
+OData defines various query options that clients can add to an OData request URL to query and transform the response OData. Here are some of the common query options:
+- `$filter` allows you to filter results based on specific conditions: `/Products?$filter=Price lt 1000 and Category eq 'Electronics'` will only return electronic products whose price is less than 1,000.
+- `$select` allows you to limit the fields that are included in the response: `/Products?$select=Name,Category,DateCreated` will only include the `Name`, `Category`, `DateCreated` fields in the response
+- `$expand` allows you to join and include related data in the response: `/OrderItems?$expand=Product` will return order items and include the related product in the response
+- `$skip` and `$top` allow you to limit the number of records in the response and can be used for client-driven pagination: `/Products$top=10&$skip=20&$orderby=Name` returns the third page of 10 products sorted by name.
+- `$apply` allows you to apply aggregations: `/Orders?$apply=groupby(Product/Name, aggregate($count as Count))` returns the number of orders per product name.
+
+These query options are not enabled by default. You can enable and configure them individual through the options passed to the `AddOData()` method:
+
+```c#
+services.AddControllers().AddOData(
+    options => options.Select().Filter().Expand().SetMaxTop(100).AddRouteComponents(edmModel));
+```
+In this example, we only enable the `$select`, `$filter`, `$expand`, and `$top` query options and we're setting the max allowable value for `$top` to be 100. If the client tries to use an OData query option that's not enabled, the service will return an error.
+
+You can enable all supported query options in one go using the `EnableQueryFeatures` method:
+
+```c#
+services.AddControllers().AddOData(
+    options => options.EnableQueryFeatures(100).AddRouteComponents(edmModel));
+```
+
+The argument passed to `EnableQueryFeatures` is the max top value. You can set it to `null` if you don't want to impose a limit on the value passed to `$top`.
+
+After enabling query options on the service, you also have to enable it on specific endpoints where you want to support query options. You do this by adding the [`[EnableQuery]`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute) attribute on the corresponding controller actions. Without specifying `[EnableQuery]` explicitly, OData query options will be ignored for that endpoint.
```

Comment:
I've rephrased it. Also rephrased it to make it clearer that `[EnableQuery]` is one option. I haven't mentioned using `ODataQueryOptions` directly since that can be covered in the query handling docs.